### PR TITLE
Adds SIMD opcodes and feature flag

### PR DESCRIPTION
**Pull request by mathetake**
part of #484 

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -105,6 +105,12 @@ type RuntimeConfig interface {
 	// See https://github.com/WebAssembly/spec/blob/main/proposals/sign-extension-ops/Overview.md
 	WithFeatureSignExtensionOps(bool) RuntimeConfig
 
+	// WithFeatureSIMD enables the vector value type and vector instructions (aka SIMD).  This defaults to false
+	// as the feature was not in WebAssembly 1.0.
+	//
+	// See https://github.com/WebAssembly/spec/blob/main/proposals/simd/SIMD.md
+	WithFeatureSIMD(bool) RuntimeConfig
```
Comment (codefromthecrypt) [inline config.go:112]: agree SIMD is the feature even if they discuss it as Vector instructions in the 2.0 changelog

File: internal/wasm/func_validation.go [hunk 2]
```diff
@@ -1042,6 +1042,19 @@ func (m *Module) validateFunctionWithMaxStackValues(
 					valueTypeStack.push(r)
 				}
 			}
+		} else if op == OpcodeVecPrefix {
+			pc++
+			// Vector instructions come with two bytes which starts with OpcodeVecPrefix,
```
Comment (anuraaga) [inline internal/wasm/func_validation.go:1047]:
```suggestion
			// Vector instructions come with two bytes where the first byte is always OpcodeVecPrefix,
```

**Review by codefromthecrypt — APPROVED** [review]
lots of instructions :D

thanks for putting this in as it helps people know if they are using the only feature in 2.0 not fully implemented.

**Review by anuraaga — COMMENTED** [review]
Just small suggestion on comment, was a bit confused reading it at first

File: internal/wasm/func_validation.go (hunk 2 above)
Comment (mathetake) [reply internal/wasm/func_validation.go:1047]: will apply your comment shortly! :D

**Pull request merged**